### PR TITLE
ui: add xss filter to id attribute

### PR DESCRIPTION
**Pull request by BATTLEHAWK00**
启用特定标签的id属性并添加过滤器，使html锚点能正常运作

### CODE REVIEW

File: packages/ui-default/backendlib/markdown-it-xss.js [hunk 1]
```diff
@@ -66,6 +66,11 @@ const xss = new Xss.FilterXSS({
     var: [],
     video: ['autoplay', 'controls', 'loop', 'preload', 'src', 'height', 'width'],
   },
+  safeAttrValue(tag, name, value) {
+    if (name === 'id') return `xss-id-${value}`;
+    if (name === 'href' && value.startsWith('#')) return `#xss-id-${value.substring(1)}`;
```
Comment (undefined-moe) [inline packages/ui-default/backendlib/markdown-it-xss.js]: `href` should not be handled as it may navigate to a built in component

**Review by undefined-moe — CHANGES_REQUESTED** [review]

File: packages/ui-default/backendlib/markdown-it-xss.js (hunk 1 above)
Comment (BATTLEHAWK00) [reply packages/ui-default/backendlib/markdown-it-xss.js]: how could it be possible? is there a built-in component that has a prefix called "xss-id"?

Comment (undefined-moe) [reply packages/ui-default/backendlib/markdown-it-xss.js]: I mean user might want to navigate to a built in anchor.

Comment (BATTLEHAWK00) [reply packages/ui-default/backendlib/markdown-it-xss.js]: all right.

**Review by undefined-moe — APPROVED** [review]

**Pull request merged**